src/Spark.Engine/Core/UriHelper.cs

```diff
@@ -68,10 +68,10 @@ public static bool HasFragment(this Uri uri)
             }
         }
 
-        public static Uri HistoryKeyFor(this IGenerator generator, Uri key)
+        public static async Task<Uri> HistoryKeyFor(this IGenerator generator, Uri key)
         {
             var identity = new ResourceIdentity(key);
-            string vid = generator.NextVersionId(identity.ResourceType, identity.Id);
+            string vid = await generator.NextVersionId(identity.ResourceType, identity.Id);
```

Comment:
This PR is on hold for this reason. I just haven't had time to return to it to add in all the ConfigureAwaits

Reply:
Andy has a good point. This library is a server-side API so there should not be any UI-thread and therefore no need to have ConfigureAwaits

Reply:
I think even server-side .Net has the issue, it's only dotnet core where it's largely unneeded. Since this project still has a demo for .Net and potential use there, I think we should add it.

Reply:
Interesting. Seems the safest is then to add ConfigureAwait